From af2c03956a5847062400cb063d9814c745566517 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:27:57 -0400
Subject: [net] netlink: Annotate RCU locking for seq_file walker

Message-id: <84605492a68e960cc0b1a7fa319bca6258068e5a.1410528971.git.jbenc@redhat.com>
Patchwork-id: 93983
O-Subject: [RHEL7.1 net PATCH 14/14] netlink: Annotate RCU locking for seq_file walker
Bugzilla: 1140661
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Flavio Leitner <fbl@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1140661
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 9ce12eb16ffb143f3a509da86283ddd0b10bcdb3
Author: Thomas Graf <tgraf@suug.ch>
Date:   Wed Aug 13 16:38:32 2014 +0200

    netlink: Annotate RCU locking for seq_file walker

    Silences the following sparse warnings:
    net/netlink/af_netlink.c:2926:21: warning: context imbalance in 'netlink_seq_start' - wrong count at exit
    net/netlink/af_netlink.c:2972:13: warning: context imbalance in 'netlink_seq_stop' - unexpected unlock

    Signed-off-by: Thomas Graf <tgraf@suug.ch>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netlink/af_netlink.c b/net/netlink/af_netlink.c
index de509b2..95dd3ba 100644
--- a/net/netlink/af_netlink.c
+++ b/net/netlink/af_netlink.c
@@ -2907,6 +2907,7 @@ static struct sock *netlink_seq_socket_idx(struct seq_file *seq, loff_t pos)
 }
 
 static void *netlink_seq_start(struct seq_file *seq, loff_t *pos)
+ __acquires(RCU)
 {
  rcu_read_lock();
  return *pos ? netlink_seq_socket_idx(seq, *pos - 1) : SEQ_START_TOKEN;
@@ -2956,6 +2957,7 @@ static void *netlink_seq_next(struct seq_file *seq, void *v, loff_t *pos)
 }
 
 static void netlink_seq_stop(struct seq_file *seq, void *v)
+ __releases(RCU)
 {
  rcu_read_unlock();
 }
-- 
1.7.1